bios/boot_sector/src/boot.s

```diff
@@ -36,13 +36,19 @@ check_int13h_extensions:
     mov bx, 0x55aa
     # dl contains drive number
     int 0x13
-    jc fail
+    jc fail_asm
     pop ax      # pop error code again
 
 rust:
     # push arguments
     push dx     # disk number
     call first_stage
+    # Fail code if first stage returns
+    push 'x'
+
+    # Call rust "fail" function that prints the character on the top of the stack
+fail_asm:
+    call fail
```

Comment:
All other labels in asm don't have a `_asm` suffix, so it seems a bit confusing to me to add one here. How about `call_fail`?

Reply:
I just removed the label jump and added conditional relative jumps to call fail. The additional asm label seemed like a lot of indirection for control flow.